### ISSUE TITLE
Make %iv_rating% etc into clickable labels

### ISSUE THREAD

**Issue by PolyglotOpenstreetmap** [Wolfsblvt/BlossomsPokemonGoManager#559]
In JOSM, my favourite OpenStreetMap editor, it's possible to click on labels to compose a search expression. Quite handy! It would also be nice to have some sort of memory for previous 'expressions' of how to rename, but that's probably a different feature request.

I tend to rename a lot. prepending e for the evolvable ones. '' for the pokémon that are ideally suited to level gyms with. ' for pokémon that are below 20cp, to fight against weak defenders between 20 and 40, which get name #%iv_rating%. This helps a lot when sorting alphabetically in the game.

I also want to express my gratitude for this great tool. I missed it a lot during the few days it wasn't available.

**Comment by Wolfsblvt** [comment]
The rename feature remembers the last pattern. Remembering more of them is planned, but not very high on the importance list.

I don't understand what you want as a feature in this issue though? :D
Care to elaborate?

**Comment by PolyglotOpenstreetmap** [comment]
Sure, I can elaborate.

I think the easiest way is to point you to: http://josm.openstreetmap.de/josm-tested.jar (It's free software, so the source code is available).

When you do Ctrl-f, you get the possibility to click on 'labels' to compose the search string.

Screenshot:
https://josm.openstreetmap.de/wiki/Help/Action/Search

**Comment by FernandoTBarros** [comment]
You can double click the labels to automatically put them in them textinput

**Comment by just-Addict** [comment]
didn't know that was supposed to be a feature, alas when I tried it, it didn't work...

EDIT: Oops, the text is a bit confusing, they _are_ copied, but not to where you expect :) Needs a Ctrl-V to actually paste them. The info text makes one think they get copied to the input...

**Comment by PolyglotOpenstreetmap** [comment]
In JOSM (The OpenStreetMap editor) you can, For BPGM it's a feature request.

Jo

2017-03-05 2:09 GMT+01:00 just-Addict <notifications@github.com>:

**Comment by just-Addict** [comment]
You already mentioned that, and FernandoTBarras commented that it already is possible in BPGM using double click, even says so above the list with patterns, my comment referred to his...as it turns out, they are dbl-clickable but the text is a bit confusing, they _are_ copied, but not to the input field as you might expect but to the clipboard. Just do a Ctrl-V in the input field to get the dbl-clicked pattern pasted

Besides that, there's lots of things possible in other software. So that's no argument to make BPGM do the same even though it would be useful. I'd rather see it add an MRU option than this. Even though it's fairly easy to have a text file with several of your most used rename patterns ready to paste in the inputfield

**Comment by PolyglotOpenstreetmap** [comment]
After having read his comment I tried it. I hadn't expected it to be copied to the clipboard. That hidden feature works for me, so thanks for clearing it up.
Of course it's true that lots of thing are possible in other software. This was merely a suggestion of something I'd think would be useful. I actually do have a text file with the more common ones I'm using...

**Comment by FernandoTBarros** [comment]
Yes sorry, I confused about this. The easier way is the way it's implemented... Other things like automatically put in the text field is a little bit trick with Swing and the way it's done. 
But if you read the instructions at the top of the window I guess is pretty clear to me:
![image](https://cloud.githubusercontent.com/assets/3437672/23608743/93c3c572-0249-11e7-9373-46bed674bcbc.png)

**Comment by Wolfsblvt** [comment]
Thanks for reporting this and for using BlossomsPokemonGoManager. This project is no longer maintained and no longer works with current Pokémon Go services, so I’m closing all remaining issues as part of archiving it. Please do not use it with a live account or current credentials. See the [Disclaimer](https://github.com/Wolfsblvt/BlossomsPokemonGoManager#disclaimer) for details.

**Comment by Wolfsblvt** [comment]
Thanks for reporting this and for using the project. Blossoms Pokémon Go Manager is no longer maintained and no longer works with current Pokémon Go services, so I’m closing all remaining issues as part of archiving the repository. This avoids leaving a false impression that further development or support is planned.
